headers/functions/arm9.h

```diff
@@ -968,6 +968,13 @@ void EnableVramBanksInSet(struct vram_banks_set* vram_banks);
 void GeomMtxLoad4x3(struct matrix_4x3* matrix);
 void GeomMtxMult4x3(struct matrix_4x3* matrix);
 void GeomGxFifoSendMtx4x3(struct matrix_4x3* matrix, void* gxfifo);
+int IncrementThreadCount(void);
+void InsertThreadOnList(struct thread* thread);
+void ThreadStart(struct thread* thread, void* function, struct thread* param_3, void* stack_area,
```

Comment:
Is the entry function always have no params? Can we define a function pointer type?

```c
typedef void (*thread_entry_fn_t)(void);
```

Reply:
Let's do

```c
typedef void (*thread_entry_fn_t)(void);
typedef void (*thread_exit_fn_t)(void);

// Contains information about a running thread
struct thread {
    ...

Reply:
Changed
Hopefully I defined them right, this C syntax is so counterintuitive to me.

Reply:
Yeah function pointer syntax is notorious for being confusing. This looks correct. At least once you get the typedef right, it's simple to use.